### PR TITLE
feat(ui): add preloader

### LINKED ISSUE #11: [UI]: Add a preloader to website
#### Description
Add a suitable preloader to the website using Sass and Javascript. The preloader should match the current color of the website.

### PR DESCRIPTION
**Pull request by PranshuNayak**
**Description**

- Added the preloader component on home page
- Added error handling in case the api endpoint request results in some error

**Related issue(s)**

Resolves #11

### CODE REVIEW

**Comment by PranshuNayak** [comment]
@akshatnema made the pr

File: web/src/components/PreLoader/PreLoader.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+function PreLoader() {
+  return (
+    <div class="flex content-center justify-center h-screen flex-wrap h-48">
+  <div class="w-1/3 p-2">
+    <div class=" text-center p-2">
+    <div className="sweet-loading h-50">
+
+        <HashLoader
+          color="#2b6cb0"
```
Comment (akshatnema) [inline web/src/components/PreLoader/PreLoader.jsx]:
```suggestion
          color="#1D4ED8"
```

Add the appropriate color to the HashLoader

File: web/src/components/PreLoader/PreLoader.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import HashLoader from "react-spinners/HashLoader";
+
+const override = {
+  display: "block",
+  margin: "0 auto",
+  borderColor: "red",
```
Comment (akshatnema) [inline web/src/components/PreLoader/PreLoader.jsx]: kindly change the border color to blue or dark blue.

File: web/src/components/Errors/index.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.server-error{
```
Comment (akshatnema) [inline web/src/components/Errors/index.css]: Don't create a separate file for the component. Use `styles/global.css` file to add this CSS code.

**Review by akshatnema — CHANGES_REQUESTED** [review]

**Comment by PranshuNayak** [comment]
@akshatnema  made the changes

**Pull request merged**